src/Microsoft.DotNet.Build.Tasks/PackageFiles/BuildVersion.targets

```diff
@@ -1,7 +1,9 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   
   <PropertyGroup>
-    <TodayTimeStamp>$([System.DateTime]::Now.ToString(yyyyMMdd))</TodayTimeStamp>
+    <!-- If OfficialBuildId is passed in, then use it as Today's TimeStamp, else default to today -->
+    <TodayTimeStamp Condition="'$(OfficialBuildId)'!=''">$(OfficialBuildId)</TodayTimeStamp>
```

Comment:
Actually, I think this gets set in all steps, though not explicitly.  Some steps may explicitly pass this variable, but in general, VSO passes all variables as environment variables, which msbuild would then pick up.   This variable gets set by the master build definition and then appears in all of the participating build definitions.  I don't think this would need to be added to any of the build definitions in order to work.